### PR TITLE
adding signal type

### PR DESCRIPTION
**Pull request by aviled**

### CODE REVIEW

File: src/SDK/SmartDetectorsShared/Clients/IInternalAnalysisServicesFactory.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    /// <summary>
+    /// An interface that extends the <see cref="IAnalysisServicesFactory"/> with methods to be used internally by the monitoring appliance
+    /// </summary>
+    public interface IInternalAnalysisServicesFactory : IAnalysisServicesFactory
+    {
+        /// <summary>
+        /// Gets the signal type based on the clients usages
+        /// </summary>
+        /// <returns>A <see cref="SignalType"/> based on the clients usages</returns>
+        SignalType GetSignalType();
```
Comment (EranGon) [inline src/SDK/SmartDetectorsShared/Clients/IInternalAnalysisServicesFactory.cs]: a. make a property
b. maybe expose two properties for UsedLogAnalysisClient and UsedMetricsClient and move the decision of which signal type to use to the analysis code?

Comment (aviled) [reply src/SDK/SmartDetectorsShared/Clients/IInternalAnalysisServicesFactory.cs]: b sounds good, changing

---
In reply to: [180400172](https://github.com/liadtal/Smart-Alerts/pull/123#discussion_r180400172) [](ancestors = 180400172)

File: src/SDK/SmartDetectorsShared/Clients/IInternalAnalysisServicesFactory.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+//        Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.Azure.Monitoring.SmartDetectors.Clients
+{
+    /// <summary>
+    /// An interface that extends the <see cref="IAnalysisServicesFactory"/> with methods to be used internally by the monitoring appliance
+    /// </summary>
+    public interface IInternalAnalysisServicesFactory : IAnalysisServicesFactory
```
Comment (EranGon) [inline src/SDK/SmartDetectorsShared/Clients/IInternalAnalysisServicesFactory.cs]: >IInternalAnalysisServicesFactory [](start = 21, length = 32)

ok, now that the properties are pretty general, you can merge the interfaces :) there's no need for an internal one

File: src/SDK/SmartDetectorsShared/Presentation/AlertExtensions.cs [hunk 3]
```diff
@@ -102,10 +107,27 @@ public static ContractsAlert CreateContractsAlert(this Alert alert, SmartDetecto
...
+        /// <param name="usedMetricClient">Indicates whether a metric client was used to create the alert</param>
+        /// <returns>A <see cref="SignalType"/> based on the clients used to create the alert</returns>
+        private static SignalType GetSignalType(bool usedLogAnalysisClient, bool usedMetricClient)
+        {
+            if (usedMetricClient)
+            {
+                return usedLogAnalysisClient ? SignalType.Multiple : SignalType.Metric;
+            }
+
+            return SignalType.Log;
```
Comment (EranGon) [inline src/SDK/SmartDetectorsShared/Presentation/AlertExtensions.cs]: >Log [](start = 30, length = 3)

multiple should be the default value

File: src/SDK/SmartDetectorsShared/Presentation/AlertExtensions.cs [hunk 4]
```diff
@@ -102,10 +107,27 @@ public static ContractsAlert CreateContractsAlert(this Alert alert, SmartDetecto
...
             };
         }
 
+        /// <summary>
+        /// Gets the <see cref="SignalType"/> based on the clients used to create the alert
+        /// </summary>
+        /// <param name="usedLogAnalysisClient">Indicates whether a log analysis client was used to create the alert</param>
+        /// <param name="usedMetricClient">Indicates whether a metric client was used to create the alert</param>
+        /// <returns>A <see cref="SignalType"/> based on the clients used to create the alert</returns>
+        private static SignalType GetSignalType(bool usedLogAnalysisClient, bool usedMetricClient)
```
Comment (EranGon) [inline src/SDK/SmartDetectorsShared/Presentation/AlertExtensions.cs]: >GetSignalType [](start = 34, length = 13)

where's the test for this?

File: src/SDK/SmartDetectorsShared/Presentation/AlertExtensions.cs [hunk 5]
```diff
@@ -102,10 +107,27 @@ public static ContractsAlert CreateContractsAlert(this Alert alert, SmartDetecto
                 AnalysisWindowSizeInMinutes = (int)request.Cadence.TotalMinutes,
                 Properties = alertProperties,
                 RawProperties = rawProperties,
-                QueryRunInfo = queryRunInfo
+                QueryRunInfo = queryRunInfo,
+                SignalType = signalType
```
Comment (EranGon) [inline src/SDK/SmartDetectorsShared/Presentation/AlertExtensions.cs]: >SignalType = signalType [](start = 16, length = 23)

where do you test this?

File: src/SDK/SmartDetectorsShared/Clients/IInternalAnalysisServicesFactory.cs (hunk 2 above)
Comment (aviled) [reply src/SDK/SmartDetectorsShared/Clients/IInternalAnalysisServicesFactory.cs]: but I don't want the developer to see these properties

---
In reply to: [180658331](https://github.com/liadtal/Smart-Alerts/pull/123#discussion_r180658331) [](ancestors = 180658331)

File: src/SDK/SmartDetectorsShared/Presentation/AlertExtensions.cs (hunk 3 above)
Comment (aviled) [reply src/SDK/SmartDetectorsShared/Presentation/AlertExtensions.cs]: not sure, why?

---
In reply to: [180658489](https://github.com/liadtal/Smart-Alerts/pull/123#discussion_r180658489) [](ancestors = 180658489)

Comment (aviled) [reply src/SDK/SmartDetectorsShared/Presentation/AlertExtensions.cs]: talked to Harel, Log is the default :)

---
In reply to: [180661162](https://github.com/liadtal/Smart-Alerts/pull/123#discussion_r180661162) [](ancestors = 180661162,180658489)

File: test/analysis/SmartDetectorsAnalysisTests/AlertPresentationTests.cs [hunk 6]
```diff
@@ -28,32 +28,33 @@ public void TestInitialize()
...
-            CollectionAssert.AreEqual(new[] { "resourceId1", "resourceId2" }, presentation.QueryRunInfo.ResourceIds.ToArray(), "Unexpected resource IDs");
+        public void WhenProcessingAlertThenTheContractsAlertIsCreatedCorrectly()
+        {
+            ContractsAlert contractsAlert = this.CreateContractsAlert(new TestAlert());
+            Assert.IsTrue(contractsAlert.AnalysisTimestamp <= DateTime.UtcNow, "Unexpected analysis timestamp in the future");
+            Assert.IsTrue(contractsAlert.AnalysisTimestamp >= DateTime.UtcNow.AddMinutes(-1), "Unexpected analysis timestamp - too back in the past");
+            Assert.AreEqual(24 * 60, contractsAlert.AnalysisWindowSizeInMinutes, "Unexpected analysis window size");
+            Assert.AreEqual(SmartDetectorName, contractsAlert.SmartDetectorName, "Unexpected Smart Detector name");
+            Assert.AreEqual("Test title", contractsAlert.Title, "Unexpected title");
+            Assert.AreEqual(SignalType.Log, contractsAlert.SignalType, "Unexpected signal type");
```
Comment (aviled) [inline test/analysis/SmartDetectorsAnalysisTests/AlertPresentationTests.cs]: >Assert.AreEqual(SignalType.Log, contractsAlert.SignalType, "Unexpected signal type"); [](start = 12, length = 85)

this is the only line added.
All other rows are just a rename of presentation to contractsAlert

File: src/SDK/SmartDetectorsShared/Clients/IInternalAnalysisServicesFactory.cs (hunk 2 above)
Comment (aviled) [reply src/SDK/SmartDetectorsShared/Clients/IInternalAnalysisServicesFactory.cs]: implemtation detail, don't want detectors using these

---
In reply to: [180670292](https://github.com/liadtal/Smart-Alerts/pull/123#discussion_r180670292) [](ancestors = 180670292,180661114,180658331)

**Pull request merged**